### PR TITLE
Documentation cleanup

### PR DESCRIPTION
**Pull request by zorkow**
This cleans up the majority of the jsdoc documentation.

There are still a few `jsdoc` linter warnings, that have to do with 
* missing `returns` type. These are sometimes too awkward to include, when they are major template types. We should have a think about them.
* some of the parser methods not always returning a result. These needs to be corrected, with accompanying tests.

Also removes unused `.eslintrc` file and minimizes rule customization.

### CODE REVIEW

**Review by dpvc — APPROVED** [review]
This looks good.  Thanks for doing all the work on this.  I have made a few minor comments and suggestions, but otherwise, ready to go.

File: ts/input/tex/textmacros/TextMacrosConfiguration.ts [hunk 1]
```diff
@@ -60,15 +60,15 @@ export const TextBaseConfiguration = Configuration.create('text-base', {
...
           'MathMacro',
           '%1 is only supported in math mode',
           '\\' + name
         );
       }
-      texParser.parse('macro', [parser, name]);
+      texParser.parse(HandlerType.MACRO, [parser, name]);
     },
   },
   items: {
```
[the remark this answers is not in the document]
Comment (zorkow) [reply ts/input/tex/textmacros/TextMacrosConfiguration.ts:74]: Both changed.

File: ts/output/chtml/DynamicFonts.ts [hunk 2]
```diff
@@ -28,11 +28,12 @@ import { ChtmlCharMap, ChtmlCharData } from './FontData.js';
  *
  * @param {{[variant: string]: {[id: string]: ChtmlCharMap}}} ranges  The variant data to be added
  * @param {string?} prefix  The prefix for when this is used from a font extension
+ * @returns {{ [variant: string]: ChtmlCharMap }} The variant map with integrated ranges
```
[the remark this answers is not in the document]
Comment (zorkow) [reply ts/output/chtml/DynamicFonts.ts:31]: There is a named type for these kind of objects, it is called `Map<...>`. There are probably quite a number of these objects that we should convert to maps, while others make more sense as dictionary style objects. 
At some point we should have a big cleanup of those.

File: ts/input/mathml/MathMLCompile.ts [hunk 3]
```diff
@@ -247,15 +247,17 @@ export class MathMLCompile<N, T, D> {
    *
    * @param {string} _name  The name of an attribute to filter
    * @param {string} value  The value to filter
+   * @returns {string} The filtered value.
    */
-  protected filterAttribute(_name: string, value: string) {
+  protected filterAttribute(_name: string, value: string): string {
```
[the remark this answers is not in the document]
Comment (zorkow) [reply ts/input/mathml/MathMLCompile.ts:252]: I thought they are overwritten in `ui/safe` extension.

[the remark this answers is not in the document]
Comment (zorkow) [reply ts/input/mathml/MathMLCompile.ts:252]: The lines here and following:

https://github.com/mathjax/MathJax-src/blob/844ad7513005fa096a07d2468bf59c7f6a6c3a50/ts/ui/safe/SafeHandler.ts#L94

File: ts/output/common/Wrappers/TeXAtom.ts [hunk 4]
```diff
@@ -104,7 +104,8 @@ export interface CommonTeXAtomClass<
 /**
  * The CommonTeXAtom wrapper mixin for the TeXAtom object
  *
- * @param Base
+ * @param {CommonWrapperConstructor} Base The constructor class
```
[the remark this answers is not in the document]
Comment (zorkow) [reply ts/output/common/Wrappers/TeXAtom.ts]: Done. I think I caught them all in commit [7eb75aa](https://github.com/mathjax/MathJax-src/pull/1135/commits/7eb75aa7a8ae68eaec6f434a3a6a995f8d89452b)

File: ts/output/common/Wrappers/XmlNode.ts [hunk 5]
```diff
@@ -276,8 +277,10 @@ export function CommonXmlNodeMixin<
 
     /**
      * The font-size and font-family values to use for the XML
+     *
+     * @returns {{'font-family': string, 'font-size': string}} The font info
```
[the remark this answers is not in the document]
Comment (zorkow) [reply ts/output/common/Wrappers/XmlNode.ts:281]: Done.

File: ts/input/mathml/MathMLCompile.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dpvc) [reply ts/input/mathml/MathMLCompile.ts:252]: Hmm.  I thought I had searched for them, but mist have done it incorrectly.  Thanks for finding them.

**Comment by zorkow** [comment]
Took care of all your points, with the exception of `filterAttribute` and `filterClassList`. If they can be removed we can do it in a separate PR.

**Pull request merged**